### DISCUSSION TITLE
Consider starting personal styleguide document

### DISCUSSION THREAD

**Discussion by jcbhmr** [spenserblack/spenserblack#43] [General]
I've had this idea for a while, but the recent PR discussion in https://github.com/spenserblack/actions-wiki/pull/13 about prefs and styles pushed me to actually do it.

This might be helpful when having others contribute to your spenserblack/* repos to be able to point to a document and say "follow this. don't ask questions in this repo, ask them there"

my pitiful starter example just covers the markdown links thing #42 so far lol https://github.com/jcbhmr/.github/wiki

then again maybe I'm overthinking this. 🙄

**Comment by spenserblack** [comment]
:laughing: Yeah, I should probably figure out exactly what style I like. Right now it's sort of an arbitrary "this feels right". Though I do lean towards `prettier`, except that I prefer `*emphasis*` over `_emphasis_` :laughing: I'm not too picky and defer to prettier, though 🙂

**Reply by jcbhmr** [reply]
I might also include other things. For instance, @jcbhmr's conventions might be
1. I use kebab-case for repo names because it
    1. The slug is in the URL and Google prefers urls-with-dashes vs urls_with_underscores https://developers.google.com/search/docs/crawling-indexing/url-structure
    2. It makes it easier to see when blue underlined (an underscore _ gets lost in the blue underline) 👈 this might be why Google says to use dashes
2. I use Discussions for ideas and "Consider X" topics
3. I use GitHub wikis to describe _how to develop in this project_
4. I use GitHub Pages to document _how to use this project_
5. The readme is a 80/20 split of for-user/for-dev-contributor docs (this is more of a "goal")
6. 🆕 trying to _contribute to other projects_ instead of _recreating my own_
7. Use Prettier for JS, Markdown, TS, JSON, HTML & CSS formatting
8. Use shfmt for Bash formatting
9. Try not to use a redundant Markdown `#` heading on documents where there's already a "title" that is present (CONTRIBUTING.md, wiki pages, etc.)

I should probably put those in my own wiki lol

**Reply by spenserblack** [reply]
Another pro for kebab-case: <kbd>-</kbd> is easier to type that <kbd>Shift</kbd>+<kbd>-</kbd>. That's why *I* use it, at least :laughing:

While prettier covers this somewhat, you might want to have a personally preferred `.editorconfig` ([example](https://github.com/spenserblack/dotfiles/blob/1a155d07cafa38846fddc6a71b2dac1f5a0f7aac/.editorconfig)).
